src/serializeBinary.ts

```diff
@@ -9,31 +9,37 @@ export async function parse(text: string): Promise<any> {
   const objects = JSON.parse(deocodedText);
   return await fromObjects(serializers(true), objects);
 }
-export async function stringify(value: any, waitForArrayBufferView = true): Promise<string> {
-  const serialized = await toObjects(serializers(waitForArrayBufferView), value);
+export async function stringify(
+  value: any,
+  waitForArrayBufferView = true
+): Promise<string> {
+  const serialized = await toObjects(
+    serializers(waitForArrayBufferView),
+    value
+  );
   // need encodeURIComponent so binary strings are transfered properly
   const message = JSON.stringify(serialized);
   return encodeURIComponent(message);
 }
 
-
 function serializers(waitForArrayBufferView: boolean) {
   return [
     ArrayBufferSerializer,
     ArrayBufferViewSerializer(waitForArrayBufferView),
-    CryptoKeySerializer
+    CryptoKeySerializer,
   ];
 }
 
-
 const ArrayBufferSerializer: Serializer<ArrayBuffer, string> = {
   id: "ArrayBuffer",
   isType: (o: any) => o instanceof ArrayBuffer,
 
   // from https://developers.google.com/web/updates/2012/06/How-to-convert-ArrayBuffer-to-and-from-String
   // modified to use Int8Array so that we can hold odd number of bytes
   toObject: async (ab: ArrayBuffer) => {
-    return String.fromCharCode.apply(null, new Int8Array(ab));
+    return new Int8Array(ab).reduce(function (data, byte) {
```

Comment:
@fitouch according to https://www.measurethat.net/Benchmarks/Show/1219/23/arraybuffer-to-base64-string
using apply is order of magnitude faster then going char by char

I suggest you split the `ab` buffer into chunks each of MAX_BUFFER_LENGTH
and do the charCode.apply on each of these buffers and concat them

Reply:
@sirpy here's a new commit based on your suggestion, thank you